src/zkevm_specs/evm_circuit/execution/error_oog_dynamic_memory_expansion.py

```diff
@@ -0,0 +1,38 @@
+from zkevm_specs.evm_circuit.table import RW
+from zkevm_specs.util.param import GAS_COST_CREATE
+from ...util import FQ
+from ..instruction import Instruction
+from ..opcode import Opcode
+from ...util import N_BYTES_GAS
+
+
+def error_oog_dynamic_memory_expansion(instruction: Instruction):
+    # retrieve op code associated to oog constant error
+    opcode = instruction.opcode_lookup(True)
+    is_create, is_create2, is_return, is_revert = instruction.multiple_select(
+        opcode, (Opcode.CREATE, Opcode.CREATE2, Opcode.RETURN, Opcode.REVERT)
+    )
+
+    # Constrain opcode must be one of CREATE, CREATE2, RETURN or REVERT.
+    instruction.constrain_equal(is_create + is_create2 + is_return + is_revert, FQ(1))
+
+    constant_gas = 0
+    stack_offset = 0
+    if is_create + is_create2 == FQ(1):
+        constant_gas = GAS_COST_CREATE
+        stack_offset += 1
```

Comment:
I saw @rrtoledo asked for an explanation why `stack_offset += 1` for `CREATE` and `CREATE2` and that there was a link to the specs docs added in the comment. I see it's because `salt` is used only for `CREATE2`, but I wouldn't mind a comment here actually :)

Reply:
fixed in [0b8af97](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/467/commits/0b8af97c5b95860d35e13749d3333b56d04045bb)